tool/tsh/registry_windows.go

```diff
@@ -0,0 +1,76 @@
+//go:build windows
+
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"strconv"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/sys/windows/registry"
+)
+
+// getRegistryKey loads or creates a registry key handle and passes it back
+func getRegistryKey(name string) (registry.Key, error) {
+	// now check for and create the individual session key
+	reg, err := registry.OpenKey(registry.CURRENT_USER, name, registry.QUERY_VALUE|registry.CREATE_SUB_KEY|registry.SET_VALUE)
+	if err != nil {
```

Comment:
Do we know if `err` is only ever set when the key isn't found? I imagine this is still harmless for most other errors, but it would be nice if we could check for some particular error type, though that's always iffy with these syscall wrapper libraries.

Reply:
The really weird thing is that https://pkg.go.dev/golang.org/x/sys/windows/registry#OpenKey kinda reads like it _always_ returns an error: "It accepts any open key, including CURRENT_USER and others, and returns the new key and an error."

That's the actual behaviour I'm seeing as well - there's always an error returned even though everything is successful. I'm wondering whether it's something to do with the permissions I'm requesting on the key. I'll mess around with it a bit and see if I can change the behaviour.

I've special-cased `syscall.ERROR_FILE_NOT_FOUND` here as that's an expected error.